src/librustc_middle/ty/sty.rs

```diff
@@ -2093,7 +2093,9 @@ impl<'tcx> TyS<'tcx> {
         variant_index: VariantIdx,
     ) -> Option<Discr<'tcx>> {
         match self.kind {
-            TyKind::Adt(adt, _) if adt.is_enum() => {
+            TyKind::Adt(adt, _)
+                if adt.is_enum() && adt.variant_range().contains(&variant_index) =>
```

Comment:
For `adt.variants.is_empty()` this function should also just bug out.

Reply:
Fixed!